lib/bedrock.js

```diff
@@ -368,6 +368,37 @@ function _setupUncaughtExceptionHandler(logger, logPrefix) {
   }
 }
 
+function _setupSignalHandler(logger, logPrefix) {
+  const SIGNALS = {
+    /*
+      The SIGHUP (“hang-up”) signal is used to report that the
+      user’s terminal is disconnected, perhaps because a network
+      or telephone connection was broken.
+    */
+    SIGHUP: 1,
+    /*
+      The SIGINT (“program interrupt”) signal is sent
+      when the user types the INTR character (normally C-c).
+    */
+    SIGINT: 2,
+    /*
+      The SIGTERM signal is a generic signal used to cause
+      program termination. Unlike SIGKILL, this signal can
+      be blocked, handled, and ignored. It is the normal way
+      to politely ask a program to terminate.
+    */
+    SIGTERM: 15
+  };
+
+  Object.keys(SIGNALS).forEach(signal => {
+    process.on(signal, async () => {
+      logger.critical(`${logPrefix} received signal [${signal}].`);
```

Comment:
- Is `logger.critical` the best choice?  I can see arguments either way here since you don't know why the signal was received.  But it also might be odd to have critical logging messages if you send the signals intentionally.
- This could remove the signal name interpolation and use a logger `{signal}` object argument.